src/secretary/core.cljs

```diff
@@ -1,6 +1,7 @@
 (ns secretary.core
   (:require [clojure.string :as string]
-            [clojure.walk :refer [keywordize-keys]]))
+            [clojure.walk :refer [keywordize-keys]])
+  (:require-macros [secretary.core :refer [defroute]]))
```

Comment:
One of my earlier commits just had:
(:require-macros [secretary.core :as s])

That was changed during my attempts to get :refer working.  From memory, it was not necessary, but neither was it harmful.

I understand your reluctance to use :refer.  However, refer is, at the moment, tricky to get working; at least, I found it so.  I would therefore recommend that you include the short forms, while deprecating their use.  I think many users will prefer the short forms, and by leaving these examples in place, you give a known working example.  I certainly would have appreciated such an example when I was trying to get it working.